workflows/workflows/paired_sample_analysis.wdl

```diff
@@ -0,0 +1,545 @@
+            description: 'One or more of the MEME suite motif databases (*.meme)',
+            group: 'reference_genome',
+            help: 'Input one or more motif databases available from the MEME suite (https://meme-suite.org/meme/db/motifs).',
+            patterns: ["*.meme"]
+        }
+        sample_bam: {
+            description: 'Sample BAM to analyze',
+            group: 'input_genomic_data',
+            help: 'Upload BAM file',
+            patterns: ["*.bam"]
+        }
+        control_bam: {
+            description: 'One input/control BAM',
+            group: 'input_genomic_data',
+            help: 'Upload BAM file.',
+            patterns: ["*.bam"]
+        }
+        results_name: {
+            description: 'Experiment results custom name',
+            group: 'analysis_parameter',
+            help: 'Input preferred analysis results name.',
+            example: 'AllMerge_mapped'
+        }
+    }
+    call macs.macs {
+        input :
+            bamfile=sample_bam,
+            control=control_bam,
+            pvalue = "1e-9",
+            keep_dup="auto",
+            output_name = if defined(results_name) then results_name + '-p9_kd-auto' else basename(sample_bam,'.bam') + '+control-p9_kd-auto',
+            default_location = if defined(results_name) then results_name + '/PEAKS/NARROW_peaks/' + results_name + '-p9_kd-auto' else sub(basename(sample_bam),'.sorted.b.*$','') + '+control/PEAKS/NARROW_peaks/' + basename(sample_bam,'.bam') + '+control-p9_kd-auto'
+    }
+
+    call util.addreadme {
+        input :
+            default_location = if defined(results_name) then results_name + '/PEAKS' else sub(basename(sample_bam),'.sorted.b.*$','') + '+control/PEAKS'
+    }
+    
+    call macs.macs as all {
+        input :
+            bamfile=sample_bam,
+            control=control_bam,
+            pvalue = "1e-9",
+            keep_dup="all",
+            output_name = if defined(results_name) then results_name + '-p9_kd-all' else basename(sample_bam,'.bam') + '+control-p9_kd-all',
+            default_location = if defined(results_name) then results_name + '/PEAKS/NARROW_peaks/' + results_name + '-p9_kd-all' else sub(basename(sample_bam),'.sorted.b.*$','') + '+control/PEAKS/NARROW_peaks/' + basename(sample_bam,'.bam') + '+control-p9_kd-all'
+    }
+
+    call macs.macs as nomodel {
+        input :
+            bamfile=sample_bam,
+            control=control_bam,
+            nomodel=true,
+            output_name = if defined(results_name) then results_name + '-nm' else basename(sample_bam,'.bam') + '+control-nm',
+            default_location = if defined(results_name) then results_name + '/PEAKS/NARROW_peaks/' + results_name + '-nm' else sub(basename(sample_bam),'.sorted.b.*$','') + '+control/PEAKS/NARROW_peaks/' + basename(sample_bam,'.bam') + '+control-nm'
+    }
+
+    call bamtogff.bamtogff {
+        input :
+            gtffile=gtf,
+            chromsizes=chromsizes,
+            bamfile=sample_bam,
```

Comment:
We use a different BAM for this step (specified above)
bamfile=s_rmdup_bam
bamindex=s_rmdup_bai

Reply:
My mistake, I assumed that was an error. Could you elaborate on why there are different BAMs used for different steps? We won't want to provide multiple BAMs for a sample and similarly won't want to provide output files that don't correspond to the BAM we do provide.

Reply:
The BAMtoGFF step provides an average read density profile in select genomic regions. We remove duplicates to eliminate for PCR/sequencing bias and to get an accurate consensus of where the antibody preferentially binds to.

Reply:
OK, thanks. I modified the workflow to pass both.